src/components/app/details/cdDetails/DeploymentDetailSteps.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react'
+import { useRouteMatch, useParams } from 'react-router'
+import { Progressing } from '../../../common'
+import { getDeploymentStatusDetail } from '../appDetails/appDetails.service'
+import { DeploymentStatusDetailsBreakdownDataType } from '../appDetails/appDetails.type'
+import DeploymentStatusDetailBreakdown from '../appDetails/DeploymentStatusBreakdown'
+import { processDeploymentStatusDetailsData } from '../appDetails/utils'
+export default function DeploymentDetailSteps() {
+    const match = useRouteMatch()
+    const { appId, envId, triggerId } = useParams<{ appId: string; envId?: string; triggerId?: string }>()
+    const [deploymentListLoader, setDeploymentListLoader] = useState<boolean>(true)
+    const [deploymentStatusDetailsBreakdownData, setDeploymentStatusDetailsBreakdownData] =
+        useState<DeploymentStatusDetailsBreakdownDataType>(processDeploymentStatusDetailsData())
+
+    let initTimer = null
+    const getDeploymentDetailStepsData = (): void => {
+        getDeploymentStatusDetail(appId, envId, triggerId)
+            .then((deploymentStatusDetailRes) => {
+                const processedDeploymentStatusDetailsData = processDeploymentStatusDetailsData(
+                    deploymentStatusDetailRes.result,
+                )
+                if (processedDeploymentStatusDetailsData.deploymentStatus === 'inprogress') {
+                    initTimer = setTimeout(() => {
+                        getDeploymentDetailStepsData()
+                    }, 10 * 1000)
+                }
+                setDeploymentStatusDetailsBreakdownData(processedDeploymentStatusDetailsData)
+                setDeploymentListLoader(false)
+            })
+            .catch((e) => {
+                setDeploymentListLoader(false)
+            })
+    }
+
+    useEffect(() => {
+        getDeploymentDetailStepsData()
+        return (): void => {
+            if (initTimer) {
+                clearTimeout(initTimer)
+            }
+        }
+    }, [])
+
+    return (
+        <>
```

Comment:
Instead of an additional diamond tag simple move the loader condition right after return something like,

```
return deploymentListLoader ? <Progressing pageLoader /> : <div>.....</div>
```